main.go

```diff
@@ -456,177 +467,170 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
+	event, err := kubeClientset.CoreV1().Events(namespace).Get(name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return nil, "not found", err
+	}
+	if err != nil {
 		log.Error().Msgf("Event %v.%v - Getting event has an error.\n\t%s", name, namespace, err.Error())
-		return "error", err
+		return nil, "error", err
 	}
 
-	return "found", nil
+	return event, "found", nil
 }
 
-func postEventAboutStatus(kubeClient *k8s.Client, secret *corev1.Secret, eventType string, action string, reason string, message string, kind string, reportingController string, reportingInstance string) (err error) {
+func postEventAboutStatus(kubeClientset *kubernetes.Clientset, secret *v1.Secret, eventType string, action string, reason string, message string, kind string, reportingController string, reportingInstance string) (err error) {
 	now := time.Now().UTC()
-	secs := int64(now.Unix())
 	count := int32(1)
-	eventName := fmt.Sprintf("%v-%v", *secret.Metadata.Name, action)
+	eventName := fmt.Sprintf("%v-%v", secret.Name, action)
 	eventSource := os.Getenv("HOSTNAME")
-	var eventResp corev1.Event
-	var exist string
-	exist, err = isEventExist(kubeClient, *secret.Metadata.Namespace, eventName, &eventResp)
+	eventResp, exist, err := isEventExist(kubeClientset, secret.Namespace, eventName)
 
 	if exist == "error" {
 		return err
 	}
 
 	if exist == "found" {
-		count = *eventResp.Count + 1
-		eventResp.Type = &eventType
-		eventResp.Action = &action
-		eventResp.Reason = &reason
-		eventResp.Message = &message
-		eventResp.Count = &count
-		eventResp.LastTimestamp.Seconds = &secs
-		err = kubeClient.Update(context.Background(), &eventResp)
+		count = eventResp.Count + 1
+		eventResp.Type = eventType
+		eventResp.Action = action
+		eventResp.Reason = reason
+		eventResp.Message = message
+		eventResp.Count = count
+		eventResp.LastTimestamp = metav1.NewTime(now)
+		_, err = kubeClientset.CoreV1().Events(secret.Namespace).Update(eventResp)
 
 		if err != nil {
-			log.Error().Msgf("Event %v.%v - Updating Event has an error.\n\t%s", *eventResp.Metadata.Name, *eventResp.Metadata.Namespace, err.Error())
+			log.Error().Msgf("Event %v.%v - Updating Event has an error.\n\t%s", eventResp.Name, eventResp.Namespace, err.Error())
 			return err
 		}
 
-		log.Info().Msgf("Event %v.%v - has been updated successfully...", *eventResp.Metadata.Name, *eventResp.Metadata.Namespace)
+		log.Info().Msgf("Event %v.%v - has been updated successfully...", eventResp.Name, eventResp.Namespace)
 		return
 	}
 
-	event := &corev1.Event{
-		Metadata: &metav1.ObjectMeta{
-			Name:      &eventName,
-			Namespace: secret.Metadata.Namespace,
-			CreationTimestamp: &metav1.Time{
-				Seconds: &secs,
-			},
-			Labels: secret.Metadata.Labels,
+	event := &v1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              eventName,
+			Namespace:         secret.Namespace,
+			CreationTimestamp: metav1.NewTime(now),
```

Comment:
Doesn't this get set automatically?

Reply:
I didn't check that. I saw that usually a [recorder](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/recorder) is used. But I didn't want to make a big change that might break logic.